src/Platforms/PostgreSQLPlatform.php

```diff
@@ -833,6 +833,12 @@ public function getDropIndexSQL($index, $table = null)
             return $this->getDropConstraintSQL($constraintName, $table);
         }
 
+        $tableName = $table instanceof Table ? $table->getQuotedName($this) : $table;
+        if ($tableName !== null && strpos($tableName, '.') !== false) {
```

Comment:
This is a bit hard to interpret. If the passed `$table` is `null`, there shouldn't be any `$tableName`, because there's no `$table`.

Please consider this:
```php
if ($table !== null) {
    $tableName = $table instanceof Table ? $table->getQuotedName($this) : $table;
    
    if (strpos($tableName, '.') !== false) {
        // ...
    }
}
```

Reply:
At this point, all 3 ifs (the added one and 2 above) have a condition on table not being null, maybe include all of them?

Reply:
You are right, `tableName` can be null in some cases. I refactored the conditions.